### PR TITLE
Bugfix disable removing or hiding the parent node (from newReferenceNodePath)

### PR DESCRIPTION
**Pull request by danielkestler**
Currently, if one removes a node from the FlatNav, the parent node is selected – if you then click "hide" or "delete" from the toolbar, the parent node will be hidden or deleted ... which can lead to a total disaster.

This PR fixes this bug by comparing the selected node with the newReferenceNodePath and hiding the buttons if they are the same. Attention: this of course only fixes the problem if the FlatNav nodes are direct children of the newReferenceNodePath node. To fix it for more cases, maybe we should check for NodeType additionally?

### CODE REVIEW

File: Resources/Private/FlatNav/src/FlatNav.js [hunk 1]
```diff
@@ -161,12 +161,16 @@ export default class FlatNav extends Component {
...
+        const newReferenceNodePath = this.props.newReferenceNodePath;
+        const parentIsFocused = focused == newReferenceNodePath;
+
         return (
             <div className={style.pageTreeContainer}>
                 <div className={style.toolbar}>
                     {!this.props.isLoadingReferenceNodePath && (<IconButton icon="plus" onClick={this.createNode}/>)}
-                    <HideSelectedNode/>
-                    <DeleteSelectedNode/>
+                    {!parentIsFocused && (<HideSelectedNode/>)}
```
Comment (dimaip) [inline Resources/Private/FlatNav/src/FlatNav.js]: Maybe it's better to set these buttons as disabled, to be consistent with how we do in the tree toolbars?

Comment (danielkestler) [reply Resources/Private/FlatNav/src/FlatNav.js]: You're right, fixed.

**Comment by danielkestler** [comment]
Found a way better to check for the allowance of hiding and removal: check if the focused node is in the list of the current this.props.nodes array.

**Comment by dimaip** [comment]
Thanks for yet another wonderful contribution!

**Pull request merged**